cli/lib/exec/spawn.js

```diff
@@ -172,7 +172,11 @@ module.exports = {
 
         if (stdioOptions.env.ELECTRON_RUN_AS_NODE) {
           // Since we are running electron as node, we need to add an entry point file.
-          startScriptPath = path.join(state.getBinaryPkgPath(path.dirname(executable)), '..', 'index.js')
+          if (isPlatform('darwin')) {
```

Comment:
Just to clarify if this fixing the issue on darwin only?

Also how did you test this - did you build the project for darwin? Or did you just pick the correct path (no actual testing)?

I've always had problems doing this locally, but I can turn on darwin builds and generate a binary via our CI for testing.

Reply:
>Just to clarify if this fixing the issue on darwin only?

I only tested in darwin - it would be nice if we can have a full covered test via CI with Linux and Windows covered.

How I did the test:

Reply:
Right I see, good idea, that's an easy way to test it quickly. You wouldn't even need to build it, you could just modify an existing binary.

Anyway, I updated our config to generate a full, installable binary for MacOS: https://github.com/cypress-io/cypress/compare/lmiller-xiaolong/issue-23636?expand=1

It takes a while (1 hour, CI + build). Once it's completed, a bot will post on that link with the install command, and we can give it a test. Please wait a bit, thanks! I'd like to get this tested and merged by the end of the week.

Reply:
Hmm it seems the binary wasn't generated, I might have done something incorrectly. I can take a look next week (Mon 13th Nov).